target/mips/cpu.h

```diff
@@ -1476,9 +1479,36 @@ static inline bool in_kernel_mode(CPUMIPSState *env) {
 // Note: the pc does not have to be up-to-date, tb start is fine.
 // We may miss a few dumps or print too many if -dfilter is on but
 // that shouldn't really matter.
-static inline target_ulong cpu_get_recent_pc(CPUMIPSState *env) {
+static inline target_ulong cpu_get_recent_pc(CPUMIPSState *env)
+{
     return env->active_tc.PC;
 }
+
+static inline bool pc_is_current(CPUArchState *env)
+{
+#ifdef CONFIG_DEBUG_TCG
+    return env->active_tc._pc_is_current;
+#else
+    return true;
+#endif
+}
+static inline void mips_update_pc_impl(TCState *state, target_ulong pc_addr)
+{
+    state->PC = pc_addr;
+#ifdef TARGET_CHERI
+    // Note: no representability check, those are done on jump.
```

Comment:
Is that to say they've already been done by the time we get here?

Reply:
Not a particularly good comment. Generally this helper should only be called with in-bounds values so we don't need a check. In cases where it may be out of bounds we will (or at least should) be doing a check later.